php-frontend/src/main/java/org/sonar/php/tree/impl/statement/IfStatementTreeImpl.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010 SonarSource and Akram Ben Aissi
+ * sonarqube@googlegroups.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.php.tree.impl.statement;
+
+import com.google.common.collect.Iterators;
+import org.sonar.php.tree.impl.PHPTree;
+import org.sonar.php.tree.impl.lexical.InternalSyntaxToken;
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.expression.ParenthesisedExpressionTree;
+import org.sonar.plugins.php.api.tree.lexical.SyntaxToken;
+import org.sonar.plugins.php.api.tree.statement.ElseClauseTree;
+import org.sonar.plugins.php.api.tree.statement.ElseifClauseTree;
+import org.sonar.plugins.php.api.tree.statement.IfStatementTree;
+import org.sonar.plugins.php.api.tree.statement.StatementTree;
+import org.sonar.plugins.php.api.visitors.TreeVisitor;
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+
+public class IfStatementTreeImpl extends PHPTree implements IfStatementTree {
+
+  private final Kind KIND;
+
+  private final InternalSyntaxToken ifToken;
+  private final ParenthesisedExpressionTree condition;
+  private final InternalSyntaxToken colonToken;
+  private final List<StatementTree> statements;
+  private final List<ElseifClauseTree> elseifClauses;
+  private final ElseClauseTree elseClause;
+  private final InternalSyntaxToken endifToken;
+  private final InternalSyntaxToken eosToken;
+
+  public IfStatementTreeImpl(
+      InternalSyntaxToken ifToken, ParenthesisedExpressionTree condition, StatementTree statement,
+      List<ElseifClauseTree> elseifClauses, ElseClauseTree elseClause
+  ) {
+    KIND = Kind.IF_STATEMENT;
+
+    this.ifToken = ifToken;
+    this.condition = condition;
+    this.statements = Collections.singletonList(statement);
+    this.elseifClauses = elseifClauses;
+    this.elseClause = elseClause;
+
+    this.colonToken = null;
+    this.endifToken = null;
+    this.eosToken = null;
+  }
+
+  public IfStatementTreeImpl(
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Constructor has 8 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)